padguide2/padguide2.py

```diff
@@ -2275,7 +2277,80 @@ def find_monster(self, query):
 
         # couldn't find anything
         return None, "Could not find a match for: " + query, None
+    
+    
+    # implements the lookup for id2, where you are allowed to specify multiple prefixes for a card
+    # search is not as robust as standard id because you should be able to be a lot more specific just from prefixes
+    def find_monster_multiple_prefixes(self, query):
+        query = rpadutils.rmdiacritics(query).lower().strip()
+        # id search
+        if query.isdigit():
+            m = self.monster_no_na_to_named_monster.get(int(query))
+            if m is None:
+                return None, 'Looks like a monster ID but was not found', None
+            else:
+                return m, None, "ID lookup"
+        
+        # handle exact nickname match
+        if query in self.all_entries:
+            return self.all_entries[query], None, "Exact nickname"
+    
+        contains_jp = rpadutils.containsJp(query)
+        if len(query) < 2 and contains_jp:
+            return None, 'Japanese queries must be at least 2 characters', None
+        elif len(query) < 4 and not contains_jp:
+            return None, 'Your query must be at least 4 letters', None
 
+        # we want to look up only the main part of the query, and then verify that each result has the prefixes
+        # so break up the query into an array of prefixes, and a string (new_query) that will be the lookup
+        query_prefixes = []
+        parts_of_query = query.split()
+        parts_of_new_query = []
+        new_query = ''
+        for i, part in enumerate(parts_of_query):
+            if part in self.all_prefixes:
+                query_prefixes.append(part)
+            else:
+                new_query = ' '.join(parts_of_query[i:])
+                break
+        
+        # if we don't actually have multiple prefixes, then default to using the regular id lookup
+        if len(query_prefixes) < 2:
+            return self.find_monster(query)
+        
+        matches = set()
+        
+        # first try to get matches from nicknames
+        for nickname, m in self.all_entries.items():
+            if new_query in nickname:
+                matches.add(m)
+        self.remove_potential_matches_without_all_prefixes(matches, query_prefixes)
+        
+        # if we don't have any candidates yet, pick a new method
+        if not len(matches):

```

Comment:
In general, prefer to invert conditions in a way that removes indentations. In short, return early when possible, continue on as if you failed.

Reply:
You should never resolve a comment without making a comment of your own, even if the comment is just 'done'. I'm surprised github even allows this.

Reply:
oh oops. Resolved per discussion in Discord that I wanted to have the option to have cumulative matches, potentially to have a list of alternative matches output along with the best match, or just to pick the best match out of multiple options (for example pantheon lookup + other lookup at the same time), this feels more flexible